### PR TITLE
Add a pinch to paint example scene to the preview package

### PR DESCRIPTION
**Pull request by JuliaRonnebergerUL**
## Summary

Adding a minimal version of paint to the preview package to show off the pinch detector, the pinch cursor and paint ribbons

## Contributor Tasks

_These tasks are for the merge request creator to tick off when creating a merge request._

- [ ] Pair review with a member of the QA team.
- [x] Add any release testing considerations to the MR for the next release.
- [x] Consider any licensing/other legal implications for this MR e.g. notices required by any new libraries.
- [x] If this MR closes a Jira issue, make sure the fix version on the JIRA issue is set to the correct one.

## Reviewer Tasks

_Add any instructions or tasks for the reviewer such as specific test considerations before this can be merged._

[Use emojis in review threads to communicate intent and help contributors.](https://github.com/ultraleap/UnityPlugin/blob/develop/CONTRIBUTING.md#review-threads)

- [x] Code reviewed.
- [x] Non-code assets e.g. Unity assets/scenes reviewed.
- [x] Checked and agree with release testing considerations added to MR for the next release.

## Closes JIRA Issue

Closes AUR-47

### CODE REVIEW

**Comment by MaxPalmer-UH** [comment]
@JuliaRonnebergerUL Can we comment public methods please? Also can you draft documentation for this, like the Core examples? I know it's preview but would be great to get the documentation done now while it's still fresh. Thanks.

File: Packages/Tracking Preview/Examples~/PinchToPaint/Materials/White.mat [hunk 1]
```diff
@@ -0,0 +1,78 @@
+%YAML 1.1
```
Comment (JuliaRonnebergerUL) [inline Packages/Tracking Preview/Examples~/PinchToPaint/Materials/White.mat:1]: test

File: Packages/Tracking Preview/Examples~/PinchToPaint/Scripts/PaintCursor.cs [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        public PinchDetector pinchDetector;
+
+
+        [Header("Cursor Following")]
+
+        public CursorFollowType cursorFollowType = CursorFollowType.Dynamic;
+
+        public enum CursorFollowType
+        {
+            Rigid,
```
[the remark this answers is not in the document]
Comment (JuliaRonnebergerUL) [reply Packages/Tracking Preview/Examples~/PinchToPaint/Scripts/PaintCursor.cs]: I removed these to simplify the code

File: Packages/Tracking Preview/Examples~/PinchToPaint/Scripts/PinchStrokeProcessor.cs [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+            }
+
+            if (_strokeProcessor.IsActualizingStroke)
+            {
+                drawTime += Time.deltaTime;
+            }
+        }
+
+        private List<StrokePoint> _cachedStrokePoints = new List<StrokePoint>();
+        private void ProcessAddStrokePoints(List<Vector3> points, List<float> effDeltaTimes)
```
[the remark this answers is not in the document]
Comment (JuliaRonnebergerUL) [reply Packages/Tracking Preview/Examples~/PinchToPaint/Scripts/PinchStrokeProcessor.cs]: I've removed the delta times of the stroke points to simplify the code

File: Packages/Tracking Preview/Examples~/PinchToPaint/Scripts/StrokeProcessor.cs [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+            }
+
+            if (_isActualizingStroke)
+            {
+                Debug.LogError("[StrokeProcessor] Stroke already actualizing; cannot begin actualizing stroke. Call StopActualizingStroke() first.");
+                return;
+            }
+            _isActualizingStroke = true;
+            _actualizedStrokeIdx = 0;
+            _strokeOutput = new List<StrokePoint>(); // can't clear -- other objects have references to the old stroke output.
```
[the remark this answers is not in the document]
Comment (JuliaRonnebergerUL) [reply Packages/Tracking Preview/Examples~/PinchToPaint/Scripts/StrokeProcessor.cs:113]: Once the renderer is done with it, it replaces the reference with a new list. So there won't be any references to it anymore and GC should deal with, I think?

File: Packages/Tracking Preview/Examples~/PinchToPaint/Scripts/PaintCursor.cs [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+        private float _thicknessMult = 1.5F;
+        private float _radius = 0F;
+        private float _minRadius = 0.02F;
+        private float _maxRadius = 0.03F;
+
+        public Vector3 Position
+        {
+            get { return this.transform.position; }
+        }
+        public Quaternion Rotation
```
[the remark this answers is not in the document]
Comment (JuliaRonnebergerUL) [reply Packages/Tracking Preview/Examples~/PinchToPaint/Scripts/PaintCursor.cs]: I just pushed a formatting fix

File: Packages/Tracking Preview/Examples~/PinchToPaint/Scripts/PinchStrokeProcessor.cs [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+
+        private Vector3 _prevPosition;
+        private SmoothedFloat _smoothedSpeed = new SmoothedFloat();
+        [HideInInspector]
+        public float drawTime = 0f;
+
+
+        void Start()
+        {
+            _smoothedSpeed.delay = _smoothingDelay;
```
[the remark this answers is not in the document]
Comment (JuliaRonnebergerUL) [reply Packages/Tracking Preview/Examples~/PinchToPaint/Scripts/PinchStrokeProcessor.cs:66]: I just pushed a formatting fix

File: Packages/Tracking Preview/Examples~/PinchToPaint/Scripts/StrokeProcessor.cs [hunk 7]
```diff
@@ -0,0 +1,366 @@
...
+        public float thickness;
+        public Color color;
+
+    }
+
+    public class StrokeProcessor
+    {
+
+        // Stroke processing configuration
+        private int _maxMemory = 8;
```
[the remark this answers is not in the document]
Comment (JuliaRonnebergerUL) [reply Packages/Tracking Preview/Examples~/PinchToPaint/Scripts/StrokeProcessor.cs]: I renamed it to `_maxStrokeBufferSize`. Is that better?

File: Packages/Tracking Preview/Examples~/PinchToPaint/Scripts/PinchStrokeProcessor.cs [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+            var thickRibbonRenderer = rendererObj.AddComponent<ThickRibbonRenderer>();
+
+            thickRibbonRenderer._finalizedRibbonParent = _ribbonParentObject;
+            thickRibbonRenderer._ribbonMaterial = _ribbonMaterial;
+            _strokeProcessor.RegisterStrokeRenderer(thickRibbonRenderer);
+        }
+
+        void Update()
+        {
+            float angleFromCameraLookVector = Vector3.Angle(Camera.main.transform.forward, _paintCursor.transform.position - Camera.main.transform.position);
```
[the remark this answers is not in the document]
Comment (JuliaRonnebergerUL) [reply Packages/Tracking Preview/Examples~/PinchToPaint/Scripts/PinchStrokeProcessor.cs:84]: Should the docs for this go into the Example docs (https://docs.ultraleap.com/unity-api/The-Examples/index.html) and have a similar structure like docs for other example scenes? Or should they go into preview (https://docs.ultraleap.com/unity-api/Preview/index.html) and talk more about the scripts used?

[the remark this answers is not in the document]
Comment (MaxPalmer-UH) [reply Packages/Tracking Preview/Examples~/PinchToPaint/Scripts/PinchStrokeProcessor.cs:84]: Probably preview.

File: Packages/Tracking Preview/Examples~/PinchToPaint/Scripts/StrokeProcessor.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (MaxPalmer-UH) [reply Packages/Tracking Preview/Examples~/PinchToPaint/Scripts/StrokeProcessor.cs]: Yes, thanks.

**Comment by MaxPalmer-UH** [comment]
@JuliaRonnebergerUL thanks for the changes. Looking really nice. One thing to put in the docs is an explanation of how to switch pinch to paint to the left hand. Happy to merge.

**Review by MaxPalmer-UH — APPROVED** [review]
Documentation covered in another ticket.

**Pull request merged**